src/browser-lib/extra-providers.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable, NgZone, StaticProvider, Inject } from '@angular/core';
+import {
+  PlatformLocation,
+  LocationChangeEvent,
+  DOCUMENT,
+} from '@angular/common';
+import { ɵBrowserPlatformLocation } from '@angular/platform-browser';
+
+@Injectable()
+export class SingleSpaPlatformLocation extends ɵBrowserPlatformLocation {
+  private ngZone!: NgZone;
+
+  // This is a simple marker that helps us to ignore PopStateEvents
+  // that was not dispatched by the browser.
+  private skipNextPopState = false;
+
+  private onPopStateListeners: ((event: LocationChangeEvent) => void)[] = [];
+
+  destroy(): void {
```

Comment:
We need to add `this.onPopStateListeners` to this PR, too. See related https://github.com/single-spa/single-spa-angular/pull/191/commits/f5ead6ddc461c8b4a3ac0d3214d72edff8bf2090